app/models/user_role.rb

```diff
@@ -64,7 +64,7 @@ def cache_user_roles!
   end
 
   def build_user_role_cache
-    [ self.cached_user_roles.build(:user_id => owner.id, :role_id => role.id) ]
+    [ self.cached_user_roles.build(:user_id => owner_id, :role_id => role_id) ]
```

Comment:
@tbrisker This is what was missing and why the initializer failed (`.id` can't be called in the middle of a callback if the object is not saved)